LibreNMS/Alert/Xmatters.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * Xmatters API Transport
+ *
+ * @author Sigurdur Stefan <https://github.com/LoveSkylark>
+ * @copyright 2023 Sigurdur Stefan
+ * @license GPL
+ */
+
+namespace LibreNMS\Alert\Transport;
+
+use LibreNMS\Alert\Transport;
+use LibreNMS\Exceptions\AlertTransportDeliveryException;
+use LibreNMS\Util\Http;
+
+class Xmatters extends Transport
+{
+    public function deliverAlert(array $alert_data): bool
+    {
+        $url = $this->config['xmatters_url'] . '/api/integration/1/functions/' . $this->config['xmatters_function'] . '/triggers';
+
+        $device_url = \Config::get('base_url');
+        $device_groups = \DeviceCache::get($alert_data['device_id'])->groups->pluck('name');
+
+        // Build request data
+        $data = [
+            'properties' => $alert_data,
+            'recipients' => $alert_data['contacts'],
+            'priority' => $this->mapSeverityToPriority($alert_data['severity']),
+        ];
+
+        $data['properties']['event_state'] = $this->mapStateToEventState($alert_data['state']);
+        $data['properties']['instance_url'] = $device_url;
+        $data['properties']['device_url'] = ($device_url . '/device/' . $alert_data['device_id']);
+        $data['properties']['alert_url'] = ($device_url . '/device/' . $alert_data['device_id'] . '/alerts/' . $alert_data['alert_id']);
+        $data['properties']['hostname'] = $alert_data['sysName'];
+        $data['properties']['ip'] = $alert_data['hostname'];
+        $data['properties']['device_groups'] = $device_groups;
```

Comment:
These feel like custom fields applicable to your install.  Are they generically applicable?